redisinsight/api/src/modules/rdi/dto/rdi-test-connections.response.dto.ts

```diff
@@ -39,11 +39,28 @@ export class RdiTestConnectionResult {
   error?: ErrorDetails;
 }
 
+export class RdiSourcesConnectionResult {
+  @ApiProperty({ description: 'Indicates if the source is connected' })
+  @Expose()
+  connected: boolean;
+
+  @ApiProperty({ description: 'Error message if connection fails', required: false })
+  @Expose()
+  error?: string;
+}
+
 export class RdiTestConnectionsResponseDto {
   @ApiProperty({
     description: 'Sources connection results',
   })
   @Expose()
+  @Type(() => RdiSourcesConnectionResult)
+  sources: RdiSourcesConnectionResult;
+
+  @ApiProperty({
+    description: 'Targets connection results',
+  })
+  @Expose()
   @Type(() => RdiTestConnectionResult)
-  sources: Record<string, RdiTestConnectionResult>;
+  targets: Record<string, RdiTestConnectionResult>;
```

Comment:
`Record<string, RdiTestConnectionResult>;` doesn't match with what you have in transformer `@Type(() => RdiTestConnectionResult)`. Not sure how it will work.

Reply:
Awesome! It will be nice if you will create new decorator for this so everyone can reuse it in the future. 
But it could be done later in a separate task.